src/event.rs

```diff
@@ -294,6 +294,77 @@ impl Command for DisableMouseCapture {
     }
 }
 
+bitflags! {
+    /// Represents special flags that tell compatible terminals to add extra information to keyboard events.
+    ///
+    /// See https://sw.kovidgoyal.net/kitty/keyboard-protocol/#progressive-enhancement for more information.
+    #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+    pub struct KeyboardEnhancementFlags: u8 {
+        /// Represent Escape and modified keys using CSI-u sequences, so they can be unambiguously
+        /// read.
+        const DISAMBIGUATE_ESCAPE_CODES = 0b0000_0001;
+        /// Add extra events with [`KeyEvent.kind`] set to [`KeyEventKind::Repeat`] or
+        /// [`KeyEventKind::Release`] when keys are autorepeated or released.
+        const REPORT_EVENT_TYPES = 0b0000_0010;
+        /// Send [alternate keycodes](https://sw.kovidgoyal.net/kitty/keyboard-protocol/#key-codes)
+        /// in addition to the base keycode.
+        ///
+        /// *Note*: these are not yet supported by crossterm.
+        const REPORT_ALTERNATE_KEYS = 0b0000_0100;
+        /// Represent all keyboard events as CSI-u sequences. This is required to get repeat/release
+        /// events for plain-text keys.
+        const REPORT_ALL_KEYS_AS_ESCAPE_CODES = 0b0000_1000;
+        /// Send the Unicode codepoint as well as the keycode.
+        ///
+        /// *Note*: this is not yet supported by crossterm.
+        const REPORT_ASSOCIATED_TEXT = 0b0001_0000;
```

Comment:
Is there a way to make this private maybe? Or should we perhaps remove the unsupported flags. This just to prevent not working API functionality

Reply:
Commented out for now.